src/main/java/org/sonarlint/intellij/SonarApplication.java

```diff
@@ -99,7 +131,7 @@ public String getComponentName() {
 
   private static void cleanOldWorkDir() {
     Path oldWorkDir = Paths.get(PathManager.getConfigPath()).resolve("sonarlint").resolve("work");
-    if (!Files.isDirectory(oldWorkDir)) {
+    if (!oldWorkDir.toFile().isDirectory()) {
```

Comment:
You did that because of SQ issue right? The performance issue only affect JDK 8 (and maybe not the latest versions). And performance is an issue only when calling this a lot of times. I would keep the previous syntax, that is more future proof to me.